src/condition/combo.rs

```diff
@@ -0,0 +1,552 @@
+use core::time::Duration;
+
+use bevy::prelude::*;
+use log::warn;
+
+use crate::prelude::*;
+
+/**
+Sequence of actions that needs to be triggered in specific order.
+
+The condition activates for one frame before resetting the combo progress.
+
+Requires using [`SpawnRelated::spawn`] because you need to pass [`Entity`]
+for step and cancel actions.
+
+# Examples
+
+Double click:
+
+```
+use bevy::prelude::*;
+use bevy_enhanced_input::prelude::*;
+
+# let mut world = World::new();
+world.spawn((
+    Menu,
+    Actions::<Menu>::spawn(SpawnWith(|context: &mut ActionSpawner<_>| {
+        let click = context
+            .spawn((Action::<Click>::new(), bindings![MouseButton::Left]))
+            .id();
+
+        context.spawn((
+            Action::<DoubleClick>::new(),
+            Combo::default().with_step(click).with_step(click),
+        ));
+    })),
+));
+
+#[derive(InputAction)]
+#[action_output(bool)]
+struct Click;
+
+#[derive(InputAction)]
+#[action_output(bool)]
+struct DoubleClick;
+
+#[derive(Component)]
+struct Menu;
+```
+*/
+#[derive(Component, Reflect, Default, Debug, Clone)]
+pub struct Combo {
+    /// Ordered sequence of steps that define the combo.
+    ///
+    /// Each step is satisfied when its [`ComboStep::events`] occur,
+    /// and steps must be completed in order.
+    pub steps: Vec<ComboStep>,
+
+    /// Actions that can cancel the combo.
+    ///
+    /// If a cancel action matches the action from the current step, it will be ignored.
+    pub cancel_actions: Vec<CancelAction>,
+
+    /// The type of time used to advance the timer.
+    pub time_kind: TimeKind,
+
+    /// Index of the current step in the combo.
+    step_index: usize,
+
+    timer: Timer,
```

Comment:
A more descriptive name here would be nice.

Reply:
Also doc comments!

Reply:
I added doc comment, but kept the name.
I'm just not sure how else to call it. But I think with the doc comment the `timer` name works :thinking: